Gini-iOS-SDK/GINIDocumentTaskManager.m

```diff
@@ -141,7 +192,38 @@ - (BFTask *)deleteDocument:(GINIDocument *)document
          cancellationToken:(BFCancellationToken *)cancellationToken {
     NSParameterAssert([document isKindOfClass:[GINIDocument class]]);
     
-    return GINIhandleHTTPerrors([_apiManager deleteDocument:document.documentId cancellationToken:cancellationToken]);
+    return [self deleteCompositeDocumentWithId:document.documentId
+                             cancellationToken:cancellationToken];
+}
+
+- (BFTask *)deleteCompositeDocumentWithId:(NSString *)documentId
+                        cancellationToken:(BFCancellationToken *)cancellationToken {
+    NSParameterAssert([documentId isKindOfClass:[NSString class]]);
+    
+    return GINIhandleHTTPerrors([_apiManager deleteDocument:documentId cancellationToken:cancellationToken]);
+}
+
+- (BFTask *)deletePartialDocumentWithId:(NSString *)documentId cancellationToken:(BFCancellationToken *)cancellationToken {
+    NSParameterAssert([documentId isKindOfClass:[NSString class]]);
+    
+    return [[self getDocumentWithId:documentId] continueWithSuccessBlock:^id(BFTask *task) {
+        
+        GINIDocument *document = (GINIDocument*) task.result;
+        return [[self deleteDocumentsWithUrls:document.parents cancellationToken:cancellationToken] continueWithSuccessBlock: ^id(BFTask *task) {
+            return [self deleteCompositeDocumentWithId:document.documentId cancellationToken:cancellationToken];
+        }];
+    }];
+}
+
+- (BFTask *)deleteDocumentsWithUrls:(NSArray<NSString*> *)urls cancellationToken:(BFCancellationToken *)cancellationToken {
+    NSMutableArray<BFTask *>* deleteTasks = [NSMutableArray new];
+    
+    for (NSString* url in urls) {
+        NSString* documentId = [[url componentsSeparatedByString:@"/"] lastObject];
```

Comment:
It would have been nice if you could avoid code like this. It kind of makes assumptions about how URLs in the service are formed. It is a little fragile.

Reply:
I thought about this, but at some point you have to get the id from the url (since you always get the url). I didn't come with a better idea, do you have a good one?